modules/standard/Socket.md

```diff
@@ -0,0 +1,210 @@
+# Socket Module API
+
+To those familiar with the Unix socket API, the method names will feel familiar, though their usage will be somewhat simpler than the raw Unix socket API. The Module will comprise of four records which will abstract lower order functions to higher and more intutive methods for user:
+- ipAddr : Address information storage structure.
+- tcpConn : A TCP Socket with established connection
+- tcpListener : A TCP Socket listening for connections
+- udpSocket : A connection less UDP Socket
+
+## Address Information
+
+`ipAddr` serves the purpose of storing address information in a single record irrespective of the protocol used which varies the buffer required to store address.
+
+To initialize an `ipAddr` there are several available overloads and defaults.
+
+**Method**
+```python
+proc ipAddr.init(host:string = "localhost", port:uint(16) = 8000, family:IPFamily = IPFamily.IPv4)
+```
+**Parameters**
+```
+host: string - address string in dot or color notation depending on family. "127.0.0.1" or "::1"
+port: uint(16) - port number for socket
+family: IPFamily - Internet Protocol version which can be IPFamily.IPv4 or IPFamily.IPv6
+```
+
+There are two more initialzers which can take in standard IP Addresses like loopback and broadcast as `host` which are defined as constants inside the `Sys` module.
+
+**Method**
+```python
+proc ipAddr.init(host:sys_in_addr_t, port: uint(16) = 8000)
+proc ipAddr.init(host:sys_in6_addr_t, port: uint(16) = 8000)
+```
+**Parameter**
+```
+host: string - standard ip addresses
+port: uint(16) - port number for socket
+```
+
+Family is determined based on type of standard address used `sys_in_addr_t` for ipv4 and `sys_in6_addr_t` for ipv6. Details about available options can be found [here](https://www.ibm.com/docs/en/zos/2.3.0?topic=considerations-special-ip-addresses#ipv6d0131000420__specialip)
+
+
+`ipAddr` instance created will then have getters for all the three arguments i.e. `host`, `port` and `family` which dont take up any space inside the record and are fetched from [sockaddr_storage](https://www.usenix.org/legacy/publications/library/proceedings/usenix2000/freenix/full_papers/metzprotocol/metzprotocol_html/node10.html)  using [getnameinfo](https://www.ibm.com/docs/en/zos/2.4.0?topic=reference-getnameinfo-flags-returned-information-examples#getnameinfofe) internally. [sockaddr_storage] is the only thing `ipAddr` stores and takes up memory for.
```

Comment:
Is this paragraph really needed? It seems to be describing implementation rather than behavior, which isn't good interface practice. Further, The `USENIX` page cited here is marked as "legacy content" with implicit encouragement to look at more modern stuff. And some of the content on the IBM page looks IBM-specific; it would probably be better to cite `https://pubs.opengroup.org/onlinepubs/9699919799/functions/getnameinfo.html` (POSIX).

Reply:
- I too think we don't need to add details about `getnameinfo` will remove it.
- for `sockaddr_storage` i will change the url.